src/main/java/edu/unc/lib/boxc/migration/cdm/services/SipService.java

```diff
@@ -120,6 +121,15 @@ public List<MigrationSip> generateSips(SipGenerationOptions options) {
             initDependencies(options, conn);
 
             Statement stmt = conn.createStatement();
+
+            // set up work generator progress bar
+            var count = stmt.executeQuery("select COUNT(*) from " + CdmIndexService.TB_NAME
```

Comment:
Since this method is already pretty long, might be good to add a private `calculateTotalWorks` method that runs the query and returns a number.

Reply:
Okay, I think I did it correctly. Thoughts?

Reply:
okay now it's ready for re-review